pod.go

```diff
@@ -144,4 +158,16 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
 		addGauge(descPodContainerStatusReady, boolFloat64(cs.Ready), cs.Name)
 		addCounter(descPodContainerStatusRestarts, float64(cs.RestartCount), cs.Name)
 	}
+
+	nodeName := p.Spec.NodeName
```

Comment:
I wonder at which point `NodeName` is set on the PodSpec when unspecified on creation. The concern is that this might be race-y with Prometheus starting to scrape this being unset and then set on the next scrape, which would make it a new time series.